br/pkg/storage/azblob_test.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2021 PingCAP, Inc. Licensed under Apache-2.0.
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	backuppb "github.com/pingcap/kvproto/pkg/brpb"
+	"github.com/stretchr/testify/require"
+)
+
+// use shared key to access azurite
+type sharedKeyAzuriteClientBuilder struct {
+}
+
+func (b *sharedKeyAzuriteClientBuilder) GetServiceClient() (azblob.ServiceClient, error) {
+	connStr := "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;"
+	return azblob.NewServiceClientFromConnectionString(connStr, nil)
+}
+
+func (b *sharedKeyAzuriteClientBuilder) GetAccountName() string {
+	return "devstoreaccount1"
+}
+
+func checkAzuriteRunning() bool {
+	checkStatement := fmt.Sprintf("lsof -i:%d ", 10000)
+	output, _ := exec.Command("sh", "-c", checkStatement).CombinedOutput()
```

Comment:
perhaps better to try if connecting to port 10000 succeeds.

Reply:
Done. Check whether the error is connection error. PTAL